src/shogun/regression/KernelRidgeRegression.cpp

```diff
@@ -138,18 +81,21 @@ bool CKernelRidgeRegression::train_machine(CFeatures *data)
 	}
 	ASSERT(kernel && kernel->has_features())
 
-	switch (m_train_func)
+	if (m_labels->get_num_labels() != kernel->get_num_vec_rhs())
```

Comment:
What about using a setter for this, that is also called from the constructors ...The setter might also initialise the alphas
IF you want them here, take all checks out of the matrix inversion method at least